crates/circuit/src/dag_circuit.rs

```diff
@@ -4947,7 +4947,9 @@ impl DAGCircuit {
     /// another that was created from the first via
     /// [DAGCircuit::copy_empty_like].
     pub fn push_back(&mut self, py: Python, instr: PackedInstruction) -> PyResult<NodeIndex> {
-        let initial_parse = self._push_back(py, instr)?;
+        // Increment the operation count
+        self.increment_op(instr.op.name());
+        let initial_parse = self.get_classical_resources(py, instr)?;
```

Comment:
It looks to me like `self.get_classical_resources` is the only part of this method that can fail, so it might be better to increment the op count only afterwards instead. Probably not super important since if we error, the state of the `DAGCircuit` shouldn't be assumed sound. But it might be good practice at least.